src/snowflake/cli/api/project/schemas/project_definition.py

```diff
@@ -95,23 +96,45 @@ def _convert_env(
         return ProjectEnvironment(default_env=(env or {}), override_env={})
 
 
-class ProjectDefinition(_DefinitionV11):
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        self._validate(kwargs)
+class _DefinitionV20(ProjectDefinition):
+    entities: Dict = Field(
+        title="Entity definitions.",
+    )
+
+    env: Union[Dict[str, str], ProjectEnvironment, None] = Field(
+        title="Environment specification for this project.",
+        default=None,
+        validation_alias="env",
+        union_mode="smart",
+    )
+
+    @field_validator("env")
+    @classmethod
+    def _convert_env(
+        cls, env: Union[Dict, ProjectEnvironment, None]
+    ) -> ProjectEnvironment:
+        if isinstance(env, ProjectEnvironment):
+            return env
+        return ProjectEnvironment(default_env=(env or {}), override_env={})
+
+    @field_validator("entities")
+    @classmethod
+    def validate_entities(cls, entities: Dict) -> Dict:
+        # TODO Add entities validation logic
+        return entities
+
 
-    @staticmethod
-    def _validate(data: Any):
-        if not isinstance(data, dict):
-            return
-        if version := str(data.get("definition_version")):
-            version_model = _version_map.get(version)
-            if not version_model:
-                raise ValueError(
-                    f"Unknown schema version: {version}. Supported version: {_supported_version}"
-                )
-            version_model(**data)
+def get_project_definition(**data):
+    if not isinstance(data, dict):
+        return
+    if FeatureFlag.ENABLE_PDF_V2.is_enabled():
```

Comment:
To confirm, we cannot put this into the version map at the outer level of this file because feature flags won't be enabled yet, correct?

Reply:
Looks like they are parsed when this executes for the first time! Moved it to the top level.

Reply:
No, it works at the top level as well, but if the config file is corrupted it will cause the CLI to crash with an uncaught exception instead of a pretty error box.